tests/main/mount-ns/task.yaml

```diff
@@ -0,0 +1,112 @@
+    snap pack test-snapd-mountinfo.core16
+    snap install --dangerous test-snapd-mountinfo_16_all.snap
+    snap connect test-snapd-mountinfo:mount-observe
+    su root -c "snap run test-snapd-mountinfo" >PER-SNAP-16.raw.txt
+    su test -c "snap run test-snapd-mountinfo" >PER-USER-16.raw.txt
+    sudo snap remove test-snapd-mountinfo
+
+    echo "Collect mountinfo from core18-based, per-snap and per-snap, per-suer mount namespaces"
+    snap pack test-snapd-mountinfo.core18
+    snap install --dangerous test-snapd-mountinfo_18_all.snap
+    snap connect test-snapd-mountinfo:mount-observe
+    su root -c "snap run test-snapd-mountinfo" >PER-SNAP-18.raw.txt
+    su test -c "snap run test-snapd-mountinfo" >PER-USER-18.raw.txt
+    sudo snap remove test-snapd-mountinfo
+
+    echo "Transform mountinfo tables to make them deterministic"
+    deterministic-mountinfo-tool                                              -f HOST.raw.txt        >HOST.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt                           -f PER-SNAP-16.raw.txt >PER-SNAP-16.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt --ref PER-SNAP-16.raw.txt -f PER-USER-16.raw.txt >PER-USER-16.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt                           -f PER-SNAP-18.raw.txt >PER-SNAP-18.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt --ref PER-SNAP-18.raw.txt -f PER-USER-18.raw.txt >PER-USER-18.deterministic.txt
+    if snap debug sandbox-features --required confinement-options:classic; then
+        deterministic-mountinfo-tool --ref HOST.raw.txt                           -f PER-SNAP-C7.raw.txt >PER-SNAP-C7.deterministic.txt
+        deterministic-mountinfo-tool --ref HOST.raw.txt --ref PER-SNAP-C7.raw.txt -f PER-USER-C7.raw.txt >PER-USER-C7.deterministic.txt
+    fi
+execute: |
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/HOST.expected.txt" HOST.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-SNAP-16.expected.txt" PER-SNAP-16.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-USER-16.expected.txt" PER-USER-16.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-SNAP-18.expected.txt" PER-SNAP-18.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-USER-18.expected.txt" PER-USER-18.deterministic.txt
+    if snap debug sandbox-features --required confinement-options:classic; then
+        diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-SNAP-C7.expected.txt" PER-SNAP-C7.deterministic.txt
+        diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-USER-C7.expected.txt" PER-USER-C7.deterministic.txt
+    fi
+restore: |
+    rm -f ./*.raw.txt ./*.deterministic.txt
+    rm -f test-snapd-mountinfo_c7_all.snap
+    rm -f test-snapd-mountinfo_16_all.snap
+    rm -f test-snapd-mountinfo_18_all.snap
+debug: |
+    for fname in ./*.deterministic.txt; do
```

Comment:
If you don't mind line numbers, the whole loop could be replaced by:
```
grep -n '' ./*.deterministic.txt
```
OTOH perhaps the output of `diff` is enough

Reply:
About `grep`, yeah but ... nah, this is more readable IMO. As for `diff` I summarised my opinion in replies to Pawel. This is very useful for grabbing the full data set in case something goes south (diff context is not that useful. and shows only the first failure). I'm open to removing it once qemu is merged and we have "stable" passes.